### PR TITLE
Update EnvironmentValueArrayDecoder to emit empty values when present

### LINKED ISSUE #122: EnvironmentVariablesProvider: split(separator:) leads to inconsistent boolean array decoding between whitespace-only and empty elements
In EnvironmentVariablesProvider.swift:347, array decoding relies on:

`string.split(separator: separator).map { $0.trimmed() }`

This approach leads to inconsistent behavior when parsing comma-separated values of whitespace-only or empty elements.

### Current behavior

split(separator:) only creates elements when the element separated with separator is a one or more characters string. Empty string, two commas one after another, does not produces an empty element.
Additionally, trimmed() applied to a string containing only a space returns the string unchanged.

Examples:
- Input:
"true, 1,,YES" → boolDecoder receives: [true, 1, YES]
- Input:
"true, 1, , YES" → boolDecoder receives: [true, 1, , YES] -> This leads to a nil value during decoding. Currently protected by a guard, resulting in a throw

### Additional observations

```swift
", ," .components(separatedBy: ",")   // count == 3
", ," .split(separator: ",")          // count == 1
```

components(separatedBy:) behaves more homogeneously for:
- empty strings
- strings containing only whitespace

This makes it more predictable when parsing arrays of strings that are expected to represent boolean values.

### Proposed solution

Replace:

`string.split(separator: separator)`

with:
`string.components(separatedBy: separator)`

Impact / considerations

This change would affect all array decodings, not only booleans, analyze the impact across other array types handled by EnvironmentVariablesProvider

### Reproduction

SwiftFiddle playground demonstrating the behavior:
https://swiftfiddle.com/vqjd6ajtfbchtb34hzejvnakiq

This issue is a follow-up to a discussion here: https://github.com/apple/swift-configuration/pull/120#discussion_r2628528473

### PR DESCRIPTION
**Pull request by Adobels**
### Motivation

Closes #122

### Modifications

- Set `omittingEmptySubsequences` to `false` in `String.split(separator:omittingEmptySubsequences:)` when decoding a string that represents Bool, Int, or Double arrays separated by commas in `EnvironmentValuesProvider`.
- Update `CLIArgumentParser.swift:125` to use `split` instead of `components(separatedBy:)`, ensuring homogeneous usage of Swift’s splitting APIs.

### Result

A string representing an array of values separated by `","` will now emit empty values when two separators appear together while parsing `EnvironmentValuesProvider` dictionaries.  
This results in treating an empty value as an empty string and, consequently, throwing an error when parsed by individual value decoders for Int, Double, and Bool.

The additional update enforces homogeneous API usage and favors Swift’s native `split(separator:)` method.

### Test Plan

Add tests covering empty values in Int, Double, and Bool arrays.

### CODE REVIEW

File: Tests/ConfigurationTests/EnvironmentVariablesProviderTests.swift [hunk 1]
```diff
@@ -124,6 +125,45 @@ struct EnvironmentVariablesProviderTests {
         #expect(throws: ConfigError.self) { try sut.value(forKey: "BOOLY_ARRAY_THROWS_1", type: .boolArray) }
         #expect(throws: ConfigError.self) { try sut.value(forKey: "BOOLY_ARRAY_THROWS_2", type: .boolArray) }
         #expect(throws: ConfigError.self) { try sut.value(forKey: "BOOLY_ARRAY_THROWS_3", type: .boolArray) }
+        #expect(throws: ConfigError.self) { try sut.value(forKey: "BOOLY_ARRAY_THROWS_4", type: .boolArray) }
+    }
+
+    @available(Configuration 1.0, *)
+    @Test func valueForKeyOfIntArrayTypes() throws {
```
Comment (czechboy0) [inline Tests/ConfigurationTests/EnvironmentVariablesProviderTests.swift]: Please add some success cases as well here for the ints and doubles

Comment (Adobels) [reply Tests/ConfigurationTests/EnvironmentVariablesProviderTests.swift]: Done.
